src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -735,6 +741,556 @@ def set_up_scale_rule(self):
+        except Exception as e:
+            handle_non_404_exception(e)
+
+        if not self.containerapp_def:
+            raise ResourceNotFoundError("The containerapp '{}' does not exist".format(self.get_argument_name()))
+
+        self.new_containerapp["properties"] = {}
+        self.set_up_from_revision()
+
+        # Doing this while API has bug. If env var is an empty string, API doesn't return "value" even though the "value" should be an empty string
+        for container in safe_get(self.containerapp_def, "properties", "template", "containers", default=[]):
+            if "env" in container:
+                for e in container["env"]:
+                    if "value" not in e:
+                        e["value"] = ""
+
+        update_map = {}
+        update_map['scale'] = self.get_argument_min_replicas() or self.get_argument_max_replicas() or self.get_argument_scale_rule_name()
+        update_map['container'] = self._need_update_container()
+        update_map['ingress'] = self.get_argument_ingress() or self.get_argument_target_port()
+        update_map['registry'] = self.get_argument_registry_server() or self.get_argument_registry_user() or self.get_argument_registry_pass()
+
+        self.set_up_service_bindings()
+        self.set_up_unbind_service_bindings()
+
+        if self.get_argument_tags():
+            _add_or_update_tags(self.new_containerapp, self.get_argument_tags())
+
+        if self.get_argument_revision_suffix() is not None:
+            self.new_containerapp["properties"]["template"] = {} if "template" not in self.new_containerapp["properties"] else self.new_containerapp["properties"]["template"]
+            self.new_containerapp["properties"]["template"]["revisionSuffix"] = self.get_argument_revision_suffix()
+
+        if self.get_argument_termination_grace_period() is not None:
+            safe_set(self.new_containerapp, "properties", "template", "terminationGracePeriodSeconds",
+                     value=self.get_argument_termination_grace_period())
+
+        if self.get_argument_workload_profile_name():
+            self.new_containerapp["properties"]["workloadProfileName"] = self.get_argument_workload_profile_name()
+
+            parsed_managed_env = parse_resource_id(self.containerapp_def["properties"]["managedEnvironmentId"])
+            managed_env_name = parsed_managed_env['name']
+            managed_env_rg = parsed_managed_env['resource_group']
+            managed_env_info = None
+            try:
+                managed_env_info = self.get_environment_client().show(cmd=self.cmd, resource_group_name=managed_env_rg, name=managed_env_name)
+            except:  # pylint: disable=bare-except
+                pass
+
+            if not managed_env_info:
+                raise ValidationError(
+                    "Error parsing the managed environment '{}' from the specified containerapp".format(
```

Comment:
when not managed_env_info, it is because env doesn't exist or other ex in RP, should not be parsing issue, right?

Reply:
fixed, only resource not found will raise this error.